ament_cppcheck/ament_cppcheck/main.py

```diff
@@ -170,7 +170,12 @@ def main(argv=sys.argv[1:]):
             'severity': error.get('severity'),
             'msg': error.get('verbose'),
         }
-        report[filename].append(data)
+        if os.path.relpath(os.path.realpath(filename)) in report:
```

Comment:
Why `realpath`?

Reply:
Because the `filename` can be in a symlinked location.

Reply:
If the symlink is `foo.cpp -> bar.cpp` then it looks like `report['foo.cpp']` would exist because of line 163 above, but this is checking if `report['bar.cpp']` exists.

Reply:
I refactored this comparison logic in 4d5d2164d46c0efb206b93a905f10c782ae8a3d5. Using `samefile` make the logic independent of which of the arguments are relative / absolute / symlinks.